File: plugins/kol-ops-bridge/reject_tags.py
"""Controlled vocabulary for operator draft-rejection learning tags."""

from __future__ import annotations

from typing import Final

REJECT_TAGS: Final[frozenset[str]] = frozenset({
    "tone_too_salesy",
    "premature_pricing",
    "wrong_sku",
    "over_promised",
    "ignored_question",
    "wrong_language",
    "too_long",
    "factual_error",
    "other",
})

DEFAULT_REJECT_TAG: Final[str] = "other"


def normalize_reject_tags(raw: list[str] | None) -> list[str]:
    """Return deduplicated valid tags; unknown values map to ``other``."""
    if not raw:
        return [DEFAULT_REJECT_TAG]
    out: list[str] = []
    seen: set[str] = set()
    for item in raw:
        tag = str(item or "").strip().lower()
        if not tag:
            continue
        if tag not in REJECT_TAGS:
            tag = DEFAULT_REJECT_TAG
        if tag not in seen:
            seen.add(tag)
            out.append(tag)
    return out or [DEFAULT_REJECT_TAG]

File: playground/kol-ops-console/frontend/src/constants/rejectTags.ts
/** Controlled reject tags — keep in sync with bridge reject_tags.py */
export const REJECT_TAGS = [
  'tone_too_salesy',
  'premature_pricing',
  'wrong_sku',
  'over_promised',
  'ignored_question',
  'wrong_language',
  'too_long',
  'factual_error',
  'other',
] as const;

export type RejectTag = (typeof REJECT_TAGS)[number];

export const REJECT_TAG_LABELS: Record<RejectTag, string> = {
  tone_too_salesy: '语气太推销',
  premature_pricing: '过早谈价',
  wrong_sku: 'SKU/产品错误',
  over_promised: '过度承诺',
  ignored_question: '未回答问题',
  wrong_language: '语言不对',
  too_long: '太长',
  factual_error: '事实错误',
  other: '其他',
};

export type RejectCorrection = {
  tags: RejectTag[];
  note: string;
  suggested_fix: string;
};

export const EMPTY_REJECT_CORRECTION: RejectCorrection = {
  tags: ['other'],
  note: '',
  suggested_fix: '',
};
